externals/kyuubi-spark-sql-engine/src/main/scala-2.12/org/apache/kyuubi/engine/spark/repl/KyuubiSparkILoop.scala

```diff
@@ -97,6 +86,23 @@ private[spark] case class KyuubiSparkILoop private (
     }
   }
 
+  private def getAllJars(currentClassLoader: ClassLoader): Array[URL] = {
+    var classLoader: ClassLoader = currentClassLoader
+    var allJars = Array.empty[URL]
+    while (classLoader != null) {
+      classLoader match {
+        case loader: MutableURLClassLoader =>
+          allJars = loader.getURLs.filter { u =>
+            u.getProtocol == "file" && new File(u.getPath).isFile
```

Comment:
it should work well in most cases, but we may need to handle SPARK-47475 in the future

Reply:
will add comments after the UT finished.
```
            // TODO: handle SPARK-47475 since Spark 4.0.0 in the future
```

Reply:
ok